### DISCUSSION TITLE
[RPi5] Add NVMe PCIe Gen 3 Mode

### DISCUSSION THREAD

**Discussion by neversleep404** [raspiblitz/raspiblitz#4394] [Feature / App Ideas]
Based on my experience, it is safe and stable to activate 
`dtparam=pciex1_gen=3` 
in the config.txt for both the Pimoroni NVMe Base and the Pineberry Pi HatDrive. Larger data sets, such as the blockchain, could be copied faster with PCie Gen 3. The heat generation with PCIe Gen 3 enabled is comparable to that in normal mode. In my opinion, there is no additional stability benefit to justify avoiding this mode.

**Comment by neversleep404** [comment]
### WD_BLACK SN770 2 TB WDS200T3X0E on PIMORONI NVMe Base after 10 Minutes running Raspiblitz

A few numbers from a productively running Raspiblitz with CL, RTL, Mempool, etc. After running for 10 minutes each, 

**without Gen 3 mode:**

![Bildschirmfoto 2024-02-01 um 11 54 47](https://github.com/raspiblitz/raspiblitz/assets/80689652/7b9bc709-b339-4b86-9506-147dd6fe0bf9)
![Bildschirmfoto 2024-02-01 um 11 55 27](https://github.com/raspiblitz/raspiblitz/assets/80689652/3604be68-7831-41ed-959c-ca8c1ed91355)
_Temperature sensor 1 was sometimes over 70 degrees. CPU load never below 1._

**And now with Gen 3 mode:**

![Bildschirmfoto 2024-02-01 um 12 06 59](https://github.com/raspiblitz/raspiblitz/assets/80689652/84f623fa-50f3-4d80-94bd-d7d210b23409)
![Bildschirmfoto 2024-02-01 um 12 07 19](https://github.com/raspiblitz/raspiblitz/assets/80689652/f5a5af35-5d18-4247-b5d9-9396bbf293ff)
_Temperature sensor 1 was always under 70 degrees. CPU load sometimes below 1._

In summary, the CPU load is lower with Gen 3 mode, and the heat of the NVMe SSD is also reduced.

**Comment by VeniVidiViciLA** [comment]
Just got my RPi5, are you running Raspiblitz 1.10.0 with yours ?

**Comment by rootzoll** [comment]
Activating Gen3 is pushed back - see here: https://github.com/raspiblitz/raspiblitz/issues/4396#issuecomment-1937643603
